crypto/s2n_pkey.c

```diff
@@ -117,11 +117,11 @@ int s2n_pkey_free(struct s2n_pkey *key)
 {
     if (key != NULL && key->free != NULL) {
         POSIX_GUARD(key->free(key));
-    }
 
-    if (key->pkey != NULL) {
-        EVP_PKEY_free(key->pkey);
-        key->pkey = NULL;
+        if (key->pkey != NULL) {
+            EVP_PKEY_free(key->pkey);
```

Comment:
nit: It doesn't seem like key->free being set is really related to freeing key->pkey. Should these checks be separate? Or does it not really matter since we expect key->free to be set anyway?